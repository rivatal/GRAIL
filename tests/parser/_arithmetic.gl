1 + 5.4;
1 - 5.4;
1 * 5.4;
1 / 5.4;
-42;
1 + -43;
1 * 2 + 3 * 4;
1 + 2 - 3 / 4;
1 * 2 + 3;