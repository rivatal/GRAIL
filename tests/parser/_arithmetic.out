Expr(Binop(Num_Lit(1), Add, Num_Lit(5.4)));
Expr(Binop(Num_Lit(1), Sub, Num_Lit(5.4)));
Expr(Binop(Num_Lit(1), Mult, Num_Lit(5.4)));
Expr(Binop(Num_Lit(1), Div, Num_Lit(5.4)));
Expr(Unop(Sub, Num_Lit(42)));
Expr(Binop(Num_Lit(1), Add, Unop(Sub, Num_Lit(43))));
Expr(Binop(Binop(Num_Lit(1), Mult, Num_Lit(2)), Add, Binop(Num_Lit(3), Mult, Num_Lit(4))));
Expr(Binop(Binop(Num_Lit(1), Add, Num_Lit(2)), Sub, Binop(Num_Lit(3), Div, Num_Lit(4))));
Expr(Binop(Binop(Num_Lit(1), Mult, Num_Lit(2)), Add, Num_Lit(3)));
